<a href="https://colab.research.google.com/github/TeraneAgaeva/Supermarket_sales_analysis-python_pandas-/blob/main/Supermarket_sales_analysis_(python_pandas)_UKR_ipynb%22%22.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Контекст даних
Зростання супермаркетів у найбільш населених містах збільшується, а конкуренція на ринку також висока. Набір даних є вигрузкою історичних продажів компанії-мережі супермаркетів у 3х різних філіях протягом 3 місяців.

**Ціль:** Цей набір даних може бути використаний для цілей розуміння тенденцій продажів, які товари в якому місті варто продавати, а поставку яких - варто скоротити. На основі цих даних можна оцінити майбутні тенденції продажів, оцінити поведінку споживачів та оптимізувати стратегію продажів.

Здійснимо елементи аналізу в цьому напрямі.

## Опис даних

| Назва оригінальна           | Переклад назви             | Опис українською                                                                                       |
|---------------------------|--------------------------|-----------------------------------------------------------------------------------------------------|
| Invoice id                | Ідентифікатор чека       | Комп’ютерно генерований номер ідентифікації чека продажу                                             |
| Branch                    | Філія                    | Філія супермаркету (доступні 3 філії, ідентифіковані як A, B та C)                                     |
| City                      | Місто                    | Місцезнаходження суперцентрів                                                                        |
| Customer type             | Тип клієнта              | Тип клієнтів, зафіксований як Члени для клієнтів, що використовують картку члена, та Звичайні без картки |
| Gender                    | Стать                    | Стать покупця                                                                                       |
| Product line              | Лінійка продуктів        | Загальні групи категоризації товарів - Електронні аксесуари, Модні аксесуари, Їжа та напої, Здоров’я та краса, Дім і стиль життя, Спорт та подорожі |
| Unit price                | Ціна за одиницю          | Ціна кожного продукту в доларах США                                                                  |
| Quantity                  | Кількість                | Кількість продуктів, придбаних клієнтом                                                             |
| Tax                       | Податок                  | 5% податковий збір для покупця                                                                       |
| Total                     | Загальна сума            | Загальна ціна з урахуванням податку                                                                  |
| Date                      | Дата                     | Дата покупки (Дані доступні з січня 2019 по березень 2019)                                           |
| Time                      | Час                      | Час покупки (з 10:00 до 21:00)                                                                       |
| Payment                   | Оплата                   | Метод оплати, використаний клієнтом для покупки (доступні 3 методи – Готівка, Кредитна картка та Електронний гаманець) |
| COGS                      | Собівартість проданих товарів | Собівартість проданих товарів                                                                      |
| Gross margin percentage   | Відсоток валового прибутку | Відсоток валового прибутку                                                                         |
| Gross income              | Валовий дохід            | Валовий дохід                                                                                       |
| Rating                    | Рейтинг                  | Рейтинг задоволення клієнтів від загального досвіду покупок (На шкалі від 1 до 10)                    |


1. Імпортуємо бібліотеку pandas в загальноприйнятому форматі.

In [1]:
import pandas as pd

2. Зчитуємо набір даних із файлу [`supermarket_sales.csv`](https://drive.google.com/file/d/1pCbTnKr51RCMopCzYIQleyaVOjf5fdZr/view?usp=sharing) в змінну `df` типу `pandas.DataFrame`.

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
!ls drive/MyDrive/supermarket_sales.csv

drive/MyDrive/supermarket_sales.csv


In [4]:
data_path = 'drive/MyDrive/supermarket_sales.csv'
df = pd.read_csv(data_path)

**3. Проведемо первинний аналіз даних:**

Типи даних:

In [5]:
type(df)

pandas.core.frame.DataFrame

Подивимось об'єм даних:

In [6]:
rows, cols = df.shape
print(f"Набір даних містить {rows} рядків та {cols} колонок")

Набір даних містить 1000 рядків та 17 колонок


In [7]:
df.shape

(1000, 17)

Обчислимо за допомогою pandas скільки оперативної памʼяті у мегабайтах займає даний набір даних:

In [8]:
total_megabytes = df.memory_usage().sum()/(1024*1024)
print(f"Даний набір даних займає {total_megabytes:.2f} мегабайт оперативної памʼяті.")

Даний набір даних займає 0.13 мегабайт оперативної памʼяті.


За допомогою бібліотеки Pandas виведемо перші 4 і останні 4 записи у наборі даних (це дві окремі команди).

In [9]:
df.head(4)

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   

             Product line  Unit price  Quantity   Tax 5%     Total       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715   1/5/2019   
1  Electronic accessories       15.28         5   3.8200   80.2200   3/8/2019   
2      Home and lifestyle       46.33         7  16.2155  340.5255   3/3/2019   
3       Health and beauty       58.22         8  23.2880  489.0480  1/27/2019   

    Time      Payment    cogs  gross margin percentage  gross income  Rating  
0  13:08      Ewallet  522.83                 4.761905       26.1415     9.1  
1  10:29         Cash   76.40                 4.761905        3.8200     9.6  
2  13:23  Credit card  324.31                 4.761905       16.2155     7.4  
3  20:33      Ewallet  465.76                 4.761905       23.2880     8.4

In [10]:
df.tail(4)

Invoice ID Branch      City Customer type  Gender         Product line  \
996  303-96-2227      B  Mandalay        Normal  Female   Home and lifestyle   
997  727-02-1313      A    Yangon        Member    Male   Food and beverages   
998  347-56-2442      A    Yangon        Normal    Male   Home and lifestyle   
999  849-09-3807      A    Yangon        Member  Female  Fashion accessories   

     Unit price  Quantity  Tax 5%     Total       Date   Time  Payment  \
996       97.38        10  48.690  1022.490   3/2/2019  17:16  Ewallet   
997       31.84         1   1.592    33.432   2/9/2019  13:22     Cash   
998       65.82         1   3.291    69.111  2/22/2019  15:33     Cash   
999       88.34         7  30.919   649.299  2/18/2019  13:28     Cash   

       cogs  gross margin percentage  gross income  Rating  
996  973.80                 4.761905        48.690     4.4  
997   31.84                 4.761905         1.592     7.7  
998   65.82                 4.761905         3.291     4.1  
999  618.38                 4.761905        30.919     6.6

Виведемо рядки даних з індексами з 615 по 619 включно:

In [11]:
df[615:620]

Invoice ID Branch       City Customer type  Gender  \
615  509-29-3912      A     Yangon        Member  Female   
616  327-40-9673      B   Mandalay        Member    Male   
617  840-19-2096      C  Naypyitaw        Member    Male   
618  828-46-6863      A     Yangon        Member    Male   
619  641-96-3695      C  Naypyitaw        Member  Female   

               Product line  Unit price  Quantity   Tax 5%     Total  \
615       Sports and travel       38.72         9  17.4240  365.9040   
616       Sports and travel       72.60         6  21.7800  457.3800   
617  Electronic accessories       87.91         5  21.9775  461.5275   
618      Food and beverages       98.53         6  29.5590  620.7390   
619     Fashion accessories       43.46         6  13.0380  273.7980   

          Date   Time      Payment    cogs  gross margin percentage  \
615  3/20/2019  12:24      Ewallet  348.48                 4.761905   
616  1/13/2019  19:51         Cash  435.60                 4.761905   
617  3/14/2019  18:10      Ewallet  439.55                 4.761905   
618  1/23/2019  11:22  Credit card  591.18                 4.761905   
619   2/7/2019  17:55      Ewallet  260.76                 4.761905   

     gross income  Rating  
615       17.4240     4.2  
616       21.7800     6.9  
617       21.9775     4.4  
618       29.5590     4.0  
619       13.0380     8.5

In [12]:
df.loc[615:619]

Invoice ID Branch       City Customer type  Gender  \
615  509-29-3912      A     Yangon        Member  Female   
616  327-40-9673      B   Mandalay        Member    Male   
617  840-19-2096      C  Naypyitaw        Member    Male   
618  828-46-6863      A     Yangon        Member    Male   
619  641-96-3695      C  Naypyitaw        Member  Female   

               Product line  Unit price  Quantity   Tax 5%     Total  \
615       Sports and travel       38.72         9  17.4240  365.9040   
616       Sports and travel       72.60         6  21.7800  457.3800   
617  Electronic accessories       87.91         5  21.9775  461.5275   
618      Food and beverages       98.53         6  29.5590  620.7390   
619     Fashion accessories       43.46         6  13.0380  273.7980   

          Date   Time      Payment    cogs  gross margin percentage  \
615  3/20/2019  12:24      Ewallet  348.48                 4.761905   
616  1/13/2019  19:51         Cash  435.60                 4.761905   
617  3/14/2019  18:10      Ewallet  439.55                 4.761905   
618  1/23/2019  11:22  Credit card  591.18                 4.761905   
619   2/7/2019  17:55      Ewallet  260.76                 4.761905   

     gross income  Rating  
615       17.4240     4.2  
616       21.7800     6.9  
617       21.9775     4.4  
618       29.5590     4.0  
619       13.0380     8.5

Виведемо назви колонок у наборі даних разом з їхніми типами. Це можна зробити кількома способами:

In [13]:
df.dtypes # 1st variant

Invoice ID                  object
Branch                      object
City                        object
Customer type               object
Gender                      object
Product line                object
Unit price                 float64
Quantity                     int64
Tax 5%                     float64
Total                      float64
Date                        object
Time                        object
Payment                     object
cogs                       float64
gross margin percentage    float64
gross income               float64
Rating                     float64
dtype: object

In [14]:
df.info()  # 2st variant

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Invoice ID               1000 non-null   object 
 1   Branch                   1000 non-null   object 
 2   City                     1000 non-null   object 
 3   Customer type            1000 non-null   object 
 4   Gender                   1000 non-null   object 
 5   Product line             1000 non-null   object 
 6   Unit price               1000 non-null   float64
 7   Quantity                 1000 non-null   int64  
 8   Tax 5%                   1000 non-null   float64
 9   Total                    1000 non-null   float64
 10  Date                     1000 non-null   object 
 11  Time                     1000 non-null   object 
 12  Payment                  1000 non-null   object 
 13  cogs                     1000 non-null   float64
 14  gross margin percentage  

In [15]:
df.columns

Index(['Invoice ID', 'Branch', 'City', 'Customer type', 'Gender',
       'Product line', 'Unit price', 'Quantity', 'Tax 5%', 'Total', 'Date',
       'Time', 'Payment', 'cogs', 'gross margin percentage', 'gross income',
       'Rating'],
      dtype='object')

4. АНАЛІЗ ДАНИХ

1. Користуючись методом DataFrame.describe() визначимо 1, 25, 50, 75 і 99 персентилі колонки gross income. Подивимось, яке значення відповідає 75отму персентилю цієї колонки і що це означає?

In [16]:
df['gross income'].describe(percentiles=[0.01, 0.25, 0.5, 0.75, 0.99])

count    1000.000000
mean       15.379369
std        11.708825
min         0.508500
1%          0.913735
25%         5.924875
50%        12.088000
75%        22.445250
99%        45.250750
max        49.650000
Name: gross income, dtype: float64

In [17]:
info_75_percentile = df['gross income'].describe(percentiles=[0.75])['75%']
info_str = f"75% персентиль рівний {info_75_percentile:.2f}"
print(info_str)

75% персентиль рівний 22.45


Висновок1: 75тому персентилю відповідає значення 22.445250 - це значить, що дані по полю gross income на 75% складаються зі значень до 22.445250. Дані, що є більшими за межами даних 75% поля gross income.

2. Виведемо кількість інвойсів за кожним унікальним містом в наборі даних.

Інвойс - це один рядок в цьому наборі даних, ми можемо цим скористатись аби швидко отримати відповідь.

In [18]:
df['City'].value_counts()

City
Yangon       340
Mandalay     332
Naypyitaw    328
Name: count, dtype: int64

3. Виведемо, який відсоток інвойсів припадає на кожне унікальне місто.

In [19]:
df['City'].value_counts(normalize=True)

City
Yangon       0.340
Mandalay     0.332
Naypyitaw    0.328
Name: proportion, dtype: float64

In [20]:
df['City'].value_counts()/df['City'].value_counts().sum()

City
Yangon       0.340
Mandalay     0.332
Naypyitaw    0.328
Name: count, dtype: float64

4. Використовуючи фільтрацію даних, знайдемо середнє значення gross income для міст Yangon та Mandalay. В якому місті воно більше?

In [21]:
mean_gross_income_yangon = df[df['City'] == 'Yangon']['gross income'].mean()
mean_gross_income_mandalay = df[df['City'] == 'Mandalay']['gross income'].mean()
print(mean_gross_income_yangon)
print(mean_gross_income_mandalay)
if mean_gross_income_yangon > mean_gross_income_mandalay:
    print("Середній дохід в Янгоні більший, ніж в Мандалаї.")
elif mean_gross_income_yangon < mean_gross_income_mandalay:
    print("Середній дохід в Мандалаї більший, ніж в Янгоні.")
else:
    print("Середній дохід в Мандалаї та  в Янгоні однаковий.")

14.874001470588235
15.232024096385542
Середній дохід в Мандалаї більший, ніж в Янгоні.


5. Чому в одному місті gross income вище? Як підвищити прибуток в мережі супермаркетів?.

Висновок 5: Можливі причини різниці gross income по 2м містам:

Mandalay трохи перевищує середня дохідність gross income над Yangon, адже кількість покупок з товарів першої необхідності було більше здійснено у Mandalay за категоріями: Food and beverages, Health and beauty, Home and lifestyle.

Саме у Mandalay деякі з цих категорій товарів дорожчі і продано більше
Більше продано товарів першої необхідності у Mandalay про що може свідчити або більша чисельність населення, або рівень дохідності населення у цьому місті, або туристичний сезон був більш насичений саме у Mandalay.

6. Для продовження дослідження різниці у прибутках в магазинах давайте зʼясуємо, чи відрізняється маржа ('gross margin percentage') для різних інвойсів в магазині?

In [22]:
df.groupby(['Invoice ID'])['gross margin percentage'].agg(['mean', 'min', 'max']).round(3)


mean    min    max
Invoice ID                      
101-17-6199  4.762  4.762  4.762
101-81-4070  4.762  4.762  4.762
102-06-2002  4.762  4.762  4.762
102-77-2261  4.762  4.762  4.762
105-10-6182  4.762  4.762  4.762
...            ...    ...    ...
894-41-5205  4.762  4.762  4.762
895-03-6665  4.762  4.762  4.762
895-66-0685  4.762  4.762  4.762
896-34-0956  4.762  4.762  4.762
898-04-2717  4.762  4.762  4.762

[1000 rows x 3 columns]

In [23]:
df.groupby(['Product line'])['gross margin percentage'].agg(['mean', 'min', 'max']).round(3)

mean    min    max
Product line                               
Electronic accessories  4.762  4.762  4.762
Fashion accessories     4.762  4.762  4.762
Food and beverages      4.762  4.762  4.762
Health and beauty       4.762  4.762  4.762
Home and lifestyle      4.762  4.762  4.762
Sports and travel       4.762  4.762  4.762

Висновок 6: gross margin percentage не відрізняється за інвойсами та групами товарів.

7. Врахуовуючи висновок в попередньому завданні, дослідимо як продаються в цих містах різні категорії товарів. Виведемо відсотковий розпоіділ кількості інвойсів по категоріям товарів Product line в містах Yangon та Mandalay.

In [24]:
group1 = df.groupby(['Product line', 'City'])[['Invoice ID']].count()
group1

Invoice ID
Product line           City                 
Electronic accessories Mandalay           55
                       Naypyitaw          55
                       Yangon             60
Fashion accessories    Mandalay           62
                       Naypyitaw          65
                       Yangon             51
Food and beverages     Mandalay           50
                       Naypyitaw          66
                       Yangon             58
Health and beauty      Mandalay           53
                       Naypyitaw          52
                       Yangon             47
Home and lifestyle     Mandalay           50
                       Naypyitaw          45
                       Yangon             65
Sports and travel      Mandalay           62
                       Naypyitaw          45
                       Yangon             59

In [25]:
filter_1 = (df['City'] == 'Yangon')
invoice_counts = df[filter_1]['Product line'].value_counts(normalize=True)
filter_2 = (df['City'] == 'Mandalay')
invoice_counts2 = df[filter_2]['Product line'].value_counts(normalize=True)

print(invoice_counts)
print(invoice_counts2)

Product line
Home and lifestyle        0.191176
Electronic accessories    0.176471
Sports and travel         0.173529
Food and beverages        0.170588
Fashion accessories       0.150000
Health and beauty         0.138235
Name: proportion, dtype: float64
Product line
Fashion accessories       0.186747
Sports and travel         0.186747
Electronic accessories    0.165663
Health and beauty         0.159639
Food and beverages        0.150602
Home and lifestyle        0.150602
Name: proportion, dtype: float64


8. Знайдемо інвойс (рядок в даних) в категорії товарів Electronic accessories, в якому був куплений товар з найбільшою кількістю і найбільшою ціною.

In [26]:
sorted_df = df.sort_values(by=["Quantity", "Unit price"], ascending=[False, False])
filter1 = sorted_df[(sorted_df['Product line'] == 'Electronic accessories')].head(1)
print(filter1)

      Invoice ID Branch    City Customer type Gender            Product line  \
105  704-48-3927      A  Yangon        Member   Male  Electronic accessories   

     Unit price  Quantity  Tax 5%    Total       Date   Time  Payment   cogs  \
105       88.67        10  44.335  931.035  1/12/2019  14:50  Ewallet  886.7   

     gross margin percentage  gross income  Rating  
105                 4.761905        44.335     7.3  


9.  Створимо нову колонку в даних Rating Category, яка визначає категорію рейтингу: високий, середній чи низький за наступним правилом:

якщо рейтинг вищий або рівний 8 - то він високий (High)
якщо рейтинг від 6 до 8 не включаючи - то він середній (Medium)
інакше - низький (Low).

In [27]:
def Category (rating):
  if rating >= 8:
        return 'High'
  elif rating >= 6 and rating < 8:
        return 'Medium'
  else:
        return 'Low'

In [28]:
df['Rating Category'] = df['Rating'].apply(Category)

Виведемо, який відсоток даних становить кожне зі значень категорій рейтингу:

In [29]:
df['Rating Category'].value_counts(normalize=True)

Rating Category
Medium    0.356
High      0.329
Low       0.315
Name: proportion, dtype: float64

10. З допомогою функції get_hour і метода DataFrame.apply() створимо нову колонку Hour в нашому наборі даних, яка буде містити години здійснення покупки, обчислені на основі часу здійснення покупки Time.

In [32]:
time = '11:28'
def get_hour(time):
  hour = int(time.split(':')[0])
  return hour
get_hour(time)

11

In [33]:
df['Hour'] = df['Time'].apply(get_hour)

11. Виведемо кількість значень за кожною годиною покупки і дайте відповідь "О яких годинах найбільше здійснюють покупок покупці і чому на Ваш погляд саме о цих годинах? Чи це слогічно, що найбільше покупок саме в ці години?".

In [34]:
sort_df = df['Hour'].value_counts().reset_index()
sort_df

Hour  count
0     19    113
1     13    103
2     15    102
3     10    101
4     18     93
5     11     90
6     12     89
7     14     83
8     16     77
9     20     75
10    17     74

In [35]:
max_time = sort_df.iloc[0]['Hour']

print(f'О {max_time} годині в основному здійснюються покупки. Я вважаю це логічним, адже більшість людей здійснюють покупки йдучи додому після роботи')

О 19 годині в основному здійснюються покупки. Я вважаю це логічним, адже більшість людей здійснюють покупки йдучи додому після роботи


12. Виведемо середнє значення чеку (Total) для покупок в містах, наявних в даних, за допомогою функції groupby.

Виведемо кількість покупок, середнє, мінімальне та максимальне значення для загальної суми чеку (Total) в містах, наявних в даних, за допомогою функції groupby.

In [36]:
df.groupby('City')['Total'].mean()

City
Mandalay     319.872506
Naypyitaw    337.099715
Yangon       312.354031
Name: Total, dtype: float64

In [37]:
df.groupby('City')['Total'].agg(['size', 'mean', 'min', 'max' ])

size        mean      min      max
City                                         
Mandalay    332  319.872506  18.6375  1022.49
Naypyitaw   328  337.099715  10.6785  1042.65
Yangon      340  312.354031  12.6945  1039.29

Висновок 12: Ми бачимо, у місті Yangon найбільше було покупок, при цьому чек був хоч і не самий високий, але не на багато відрізняється від самого високого. Можемо зробити висновок, що купівельна спроможність у Yangon нижча ніж у інших містах, дивлячись на середній чек, але кількість жителів там більше чи така культура, що люди купують меньше, але часто. У Naypyitaw вища купівельна спроможність у жителів, але купують рідше, чи можливо графік магазину не зручний і тому треба купувати з запасом.

13. Дослідимо, який філіал супермаркету ('Branch') є найприбутковішим.

In [38]:
df1 = df.groupby('Branch')['gross income'].sum().reset_index()
df2 = df1.sort_values(by=["gross income"], ascending=[False])
df2

Branch  gross income
2      C     5265.1765
0      A     5057.1605
1      B     5057.0320

In [39]:
max_branch = df2.iloc[0]['Branch']

print(f' Філія {max_branch} найбільш прибуткова.')

 Філія C найбільш прибуткова.


14. В якому місті знайходиться філіал з найвищим прибутком? Може в тому місці нам розмітисти ще один магазин?

In [40]:
city_profit = df.groupby('City')['gross income'].sum()
city_profit

City
Mandalay     5057.0320
Naypyitaw    5265.1765
Yangon       5057.1605
Name: gross income, dtype: float64

In [41]:
# Знаходимо місто з найвищим прибутком
max_profit_city = city_profit.idxmax()

print(f"Філіал з найвищим прибутком розташований в місті: {max_profit_city}")

Філіал з найвищим прибутком розташований в місті: Naypyitaw


15. Створимо зведену таблицю, яка покаже, скільки покупок (інвойсів) було зроблено в кожній з філій (Branch) за різними категоріями товарів. Запишемо таблицю в змінну invoices_by_category і виведемо змінну на екран. Ця таблиця допоможе проаналізувати, в якій філії купують найбільше товарів кожної з категорій.

In [42]:
invoices_by_category = pd.pivot_table(df, index='Branch', columns='Product line', values='Quantity', aggfunc="count")
invoices_by_category

Product line  Electronic accessories  Fashion accessories  Food and beverages  \
Branch                                                                          
A                                 60                   51                  58   
B                                 55                   62                  50   
C                                 55                   65                  66   

Product line  Health and beauty  Home and lifestyle  Sports and travel  
Branch                                                                  
A                            47                  65                 59  
B                            53                  50                 62  
C                            52                  45                 45

16. Викристовуючи змінну invoices_by_category дамовідповідь в якому філіалі магазину (Branch) найбільше інвойсів із покупкою товарів категорії "Електронні аксесуари" (Electronic accessories)?

In [43]:
max_quantity = invoices_by_category['Electronic accessories'].idxmax()
print(f'Найбільше інвойсів із покупкою товарів категорії "Електронні аксесуари" (Electronic accessories)у філіалі магазину {max_quantity}.')

Найбільше інвойсів із покупкою товарів категорії "Електронні аксесуари" (Electronic accessories)у філіалі магазину A.


17. Який відсоток покупок сплачується карткою чи електронним гаманцем порівнянно з готівкою? Якщо безготівкова оплата більше 30%, то можливо варто розглянути можливість створення онлайн магазину?

In [44]:
payment_type = df['Payment'].value_counts(normalize=True).reset_index()
payment_type

Payment  proportion
0      Ewallet       0.345
1         Cash       0.344
2  Credit card       0.311

In [45]:
not_cash = payment_type.iloc[0]['proportion'] + payment_type.iloc[2]['proportion']
not_cash

0.6559999999999999

In [46]:
print(f'{not_cash:.2f} сплачується не готівкою, отже варіант створення он-лайн магазину - це гарна ідея, яку потрібно реалізовувати, адже у клієнтів буде можливість платити он-лайн')

0.66 сплачується не готівкою, отже варіант створення он-лайн магазину - це гарна ідея, яку потрібно реалізовувати, адже у клієнтів буде можливість платити он-лайн


18. Які товари користуються найменьшим попитом? Яка дохідність від них?

In [47]:
demand = df.groupby('Product line')[['Quantity', 'Total']].sum().reset_index()
demand

Product line  Quantity       Total
0  Electronic accessories       971  54337.5315
1     Fashion accessories       902  54305.8950
2      Food and beverages       952  56144.8440
3       Health and beauty       854  49193.7390
4      Home and lifestyle       911  53861.9130
5       Sports and travel       920  55122.8265

In [48]:
min_quantity_index = demand['Quantity'].idxmin()
min_quantity_index

3

In [49]:
min_quantity_item = demand.loc[min_quantity_index, 'Product line']
min_quantity_item

'Health and beauty'

In [50]:
min_quantity = demand.loc[min_quantity_index, 'Quantity']
min_quantity

854

In [51]:
print(f'Товари {min_quantity_item} користуються найнижчим попитом у кількості {min_quantity} покупок, але кількість покупок меньше від покупок інших товарів не кретично, потрібно ще додатково аналізувати можливо дохідність та маржу')

Товари Health and beauty користуються найнижчим попитом у кількості 854 покупок, але кількість покупок меньше від покупок інших товарів не кретично, потрібно ще додатково аналізувати можливо дохідність та маржу


19. Чи налаштована комунікація з зареєстрованими покупцями типу members, тобто чи частіше/більше вони купують ніж інші.

In [52]:
customers = df.groupby('Customer type')[['Quantity', 'Total']].sum().reset_index()
customers

Customer type  Quantity       Total
0        Member      2785  164223.444
1        Normal      2725  158743.305

In [53]:
payment_type = df['Customer type'].value_counts(normalize=True).reset_index()
payment_type

Customer type  proportion
0        Member       0.501
1        Normal       0.499

In [54]:
max_quantity_index = payment_type['proportion'].idxmax()
max_quantity_index

0

In [55]:
max_quantity_item = payment_type.loc[max_quantity_index, 'Customer type']
max_quantity_item

'Member'

In [56]:
print(demand.columns)

Index(['Product line', 'Quantity', 'Total'], dtype='object')


In [57]:
max_proportion = payment_type.loc[max_quantity_index, 'proportion']
max_proportion

0.501

In [58]:
def concludions():
  if max_quantity_item == 'Member':
    print(f'Member купують більше - це добре, так має і бути. Потрібно проаналізувати відсоток покупок {max_proportion} щоб зрозуміти чи можна збільшити цей відсоток')
  else:
    print(f'Member купують меньше - потрібно вести політику для збільшення кількості Member та збільшення купівель ними, адже це лояльність до нашої компанії постійних покупців та звязок з клієнтами. Отже потрібен додатковий аналіз для покращення ситуації')
concludions()

Member купують більше - це добре, так має і бути. Потрібно проаналізувати відсоток покупок 0.501 щоб зрозуміти чи можна збільшити цей відсоток
